src/OptimalFilter.jl

```diff
@@ -139,59 +139,64 @@ end
 
 # Normalized two-dimension discrete Fourier transofrm normalized by sqrt(n1_bar).
 # Operates on the 2d data stored as a matrix.
+# The argument `f` lets you switch between forward transform (`f=identity`) and
+# inverse transform (`f=inv`).
 # From Dietrich & Newsam 96 in text following equation 12
-function normalized_2d_fft!(transformed_array::AbstractMatrix{T}, array::AbstractMatrix{S}, grid_ext) where {T,S}
+function normalized_2d_fft!(transformed_array::AbstractMatrix{<:Complex}, array::AbstractMatrix{S}, fft_plan::FFTW.FFTWPlan, fft_plan!::FFTW.FFTWPlan, grid_ext, f=identity) where {S}
 
-    normalization_factor = 1.0 / sqrt(grid_ext.nx * grid_ext.ny)
-    transformed_array .= fft(array) .* normalization_factor
+    normalization_factor = f(sqrt(grid_ext.nx * grid_ext.ny))
+    if pointer(transformed_array) == pointer(array)
```

Comment:
What are the pointers for?

Reply:
See the answer to https://github.com/Team-RADDISH/ParticleDA.jl/pull/159#discussion_r589433961